app/Entry.js

```diff
@@ -27,127 +27,58 @@ import { SettingsScreen } from './views/Settings';
 
 const Stack = createStackNavigator();
 
-class Entry extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      initialRouteName: '',
-    };
-  }
+export const Entry = () => {
+  const [onboardingDone, setOnboardingDone] = useState(false);
 
-  componentDidMount() {
-    GetStoreData('ONBOARDING_DONE')
-      .then(onboardingDone => {
-        this.setState({
-          initialRouteName: onboardingDone,
-        });
-      })
-      .catch(error => console.log(error));
-  }
+  useEffect(() => {
+    async function checkDone() {
+      const flag = await GetStoreData('ONBOARDING_DONE');
```

Comment:
```suggestion
      const flag = await GetStoreData('ONBOARDING_DONE', false);
```

should use `JSON.parse()` to convert that to a regular boolean.

Also - maybe throw `ONBOARDING_DONE` into `constants/storage.js` ?

Reply:
Done